presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -455,6 +458,23 @@ protected TupleDescriptor visitInsert(Insert insert, AnalysisContext context)
         return new TupleDescriptor(Field.newUnqualified("rows", BIGINT));
     }
 
+    private boolean typesMatchForInsert(Iterable<Type> tableTypes, Iterable<Type> queryTypes)
```

Comment:
Not doing that.
naming method elementsEqual is misleading. As "equality" function we are passing to it is not symmetric.